include/itkAttenuationImageFilter.h

```diff
@@ -0,0 +1,204 @@
+
+  using InputRegionType = typename TInputImage::RegionType;
+  using InputSizeType = typename TInputImage::SizeType;
+  using InputIndexType = typename TInputImage::IndexType;
+  using InputPixelType = typename TInputImage::PixelType;
+
+  using OutputRegionType = typename TOutputImage::RegionType;
+  using OutputPixelType = typename TOutputImage::PixelType;
+
+  itkSetInputMacro(MaskImage, TMaskImage);
+  itkGetInputMacro(MaskImage, TMaskImage);
+
+  itkGetConstMacro(LabelValue, MaskPixelType);
+  itkSetMacro(LabelValue, MaskPixelType);
+
+  itkGetConstMacro(Direction, unsigned int);
+  itkSetMacro(Direction, unsigned int);
+
+  itkGetConstMacro(SamplingFrequencyMHz, float);
+  itkSetMacro(SamplingFrequencyMHz, float);
+
+  itkGetConstMacro(FrequencyBandStartMHz, float);
+  itkSetMacro(FrequencyBandStartMHz, float);
+
+  itkGetConstMacro(FrequencyBandEndMHz, float);
+  itkSetMacro(FrequencyBandEndMHz, float);
+
+  itkGetConstMacro(ConsiderNegativeAttenuations, bool);
+  itkSetMacro(ConsiderNegativeAttenuations, bool);
+
+  itkGetConstMacro(PadPixelUpperBounds, unsigned int);
+  itkSetMacro(PadPixelUpperBounds, unsigned int);
+  itkGetConstMacro(PadPixelLowerBounds, unsigned int);
+  itkSetMacro(PadPixelLowerBounds, unsigned int);
+
+  itkGetConstMacro(PadSpatialUpperBoundsMM, float);
+  itkSetMacro(PadSpatialUpperBoundsMM, float);
+  itkGetConstMacro(PadSpatialLowerBoundsMM, float);
+  itkSetMacro(PadSpatialLowerBoundsMM, float);
+
+  // Alias for setting direction of RF waveform in data collection
+  void
+  SetScanDirection(unsigned int direction)
+  {
+    this->SetDirection(direction);
+  };
+  unsigned int
+  GetScanDirection()
+  {
+    return this->GetDirection();
+  };
+
+  void
+  PrintSelf(std::ostream & os, Indent indent) const override;
+
+protected:
+  AttenuationImageFilter();
+  ~AttenuationImageFilter() override = default;
+
+  /** Initialize some accumulators before any chunks are processes */
+  void
+  BeforeThreadedGenerateData() override;
+
+  /** Do final mean and variance computation from data accumulated in threads.
+   */
+  void
+  AfterThreadedGenerateData() override;
+
+  void
+  DynamicThreadedGenerateData(const OutputRegionType &) override;
+
+  const ImageRegionSplitterBase *
+  GetImageRegionSplitter() const override;
+
+  /** Compute attenuation between two pixels in the RF spectra vector image.
+   *  Assumes that image spacing is in MM. */
+  OutputPixelType
+  ComputeAttenuation(const InputIndexType & end, const InputIndexType & start) const;
+
+private:
+  /** Label value indicating pixel should be included in analysis */
+  MaskPixelType m_LabelValue = 1;
+
+  /** RF scanline direction */
+  unsigned int m_Direction = 0;
+
+  /** RF sampling frequency */
+  float m_SamplingFrequencyMHz = 0.0f;
+
+  /** RF frequency band */
+  float m_FrequencyBandStartMHz = 0.0f;
+  float m_FrequencyBandEndMHz = 0.0f;
+
+  /** Optionally discard negative attenuation estimates
```

Comment:
I think these docstrings should go above setters, .e.g this one should go to line 100.